src/item.cpp

```diff
@@ -1090,13 +1090,32 @@ bool item::merge_charges( const item &rhs )
 ret_val<bool> item::put_in( const item &payload, item_pocket::pocket_type pk_type )
 {
     ret_val<bool> result = contents.insert_item( payload, pk_type );
+    if( !result.success() ) {
+        debugmsg( "tried to put an item (%s) count (%d) in a container (%s) that cannot contain it: %s",
+                  payload.typeId().str(), payload.count(), typeId().str(), result.str() );
+    }
     if( pk_type == item_pocket::pocket_type::MOD ) {
         update_modified_pockets();
     }
     on_contents_changed();
     return result;
 }
 
+ret_val<bool> item::put_in( const item &payload )
+{
+    item_pocket::pocket_type pk_type = item_pocket::pocket_type::CONTAINER;
+    if( ( is_gun() && payload.is_gunmod() ) || ( is_tool() && payload.is_toolmod() ) ) {
+        pk_type = item_pocket::pocket_type::MOD;
+    } else if( typeId().str() == "usb_drive" && payload.is_software() ) {
+        pk_type = item_pocket::pocket_type::SOFTWARE;
+    } else if( ( is_gun() || is_tool() ) && ( payload.is_battery() || payload.is_magazine() ) ) {
```

Comment:
```suggestion
    } else if( ( is_gun() || is_tool() ) && payload.is_magazine() ) {
```
battery is magazine too.